src/plugins/tests.jl

```diff
@@ -19,44 +29,106 @@ Sets up testing for packages.
 @plugin struct Tests <: FilePlugin
     file::String = default_file("test", "runtests.jl")
     project::Bool = false
+    aqua::Bool = false
+    aqua_kwargs::NamedTuple = NamedTuple()  # type-unstable
 end
 
 source(p::Tests) = p.file
 destination(::Tests) = joinpath("test", "runtests.jl")
-view(::Tests, ::Template, pkg::AbstractString) = Dict("PKG" => pkg)
+
+function view(p::Tests, ::Template, pkg::AbstractString)
+    d = Dict("PKG" => pkg)
+    if p.aqua
+        if isempty(p.aqua_kwargs)
+            kwargs_str = ""
+        else
+            kwargs_str = "; " * strip(string(p.aqua_kwargs), ['(', ')'])
+        end
+        d["AQUA_IMPORT"] = ", Aqua"
+        d["AQUA_TESTSET"] = """
+
+            @testset verbose = true "Code quality (Aqua.jl)" begin
+                Aqua.test_all($pkg$kwargs_str)
+            end
+        """[1:end-1]
+    else
+        d["AQUA_IMPORT"] = ""
+        d["AQUA_TESTSET"] = ""
+    end
+    return d
+end
 
 function validate(p::Tests, t::Template)
-    invoke(validate, Tuple{FilePlugin, Template}, p, t)
+    invoke(validate, Tuple{FilePlugin,Template}, p, t)
     p.project && t.julia < v"1.2" && @warn string(
-        "Tests: The project option is set to create a project (supported in Julia 1.2 and later) ",
-        "but a Julia version older than 1.2 ($(t.julia)) is supported by the template",
+            "Tests: The project option is set to create a project (supported in Julia 1.2 and later) ",
+            "but a Julia version older than 1.2 ($(t.julia)) is supported by the template",
+        )
+    aqua_kwargs_names = (
+        :ambiguities,
+        :unbound_args,
+        :undefined_exports,
+        :piracy,
+        :project_extras,
+        :stale_deps,
+        :deps_compat,
+        :project_toml_formatting,
     )
+    for (key, val) in pairs(p.aqua_kwargs)
+        if !(val isa Bool)
+            throw(ArgumentError("Aqua keyword arguments must have boolean values"))
+        elseif !(key in aqua_kwargs_names)
+            throw(ArgumentError("Aqua keyword arguments must belong to $aqua_kwargnames"))
+        end
+    end
```

Comment:
```suggestion
    end
    _validate_aqua_version(p.aqua_version)
```

and then outside this function:
```
_validate_aqua_version(::Nothing) = nothing
_validate_aqua_version(::VersionNumber) = nothing
function _validate_aqua_version(v::AbstractString)
    v == "*" || throw(ArgumentError("Aqua version must be `VersionNumber`, `"*"` for the latest version of Aqua.jl, or `nothing` to skip use of Aqua."
    return nothing
end

```